nemo/collections/tts/parts/preprocessing/feature_processors.py

```diff
@@ -0,0 +1,223 @@
+            stats_path: JSON file with feature mean and variance.
+            mask_field: Optional, field in example dictionary with boolean array indicating which values to
+                mask to 0. Defaults to 'voiced_mask', expected to be computed by pyin pitch estimator.
+        """
+
+        self.field = field
+        self.mask_field = mask_field
+
+        if not os.path.exists(stats_path):
+            raise ValueError(f"Statistics file does not exist: {stats_path}")
+
+        with open(stats_path, 'r', encoding="utf-8") as stats_f:
+            stats_dict = json.load(stats_f)
+            self.mean = stats_dict["default"][f"{self.field}_mean"]
+            self.std = stats_dict["default"][f"{self.field}_std"]
+
+    def process(self, training_example: dict) -> None:
+        feature = training_example[self.field]
+
+        feature = (feature - self.mean) / self.std
+        if self.mask_field:
+            voiced_mask = training_example[self.mask_field]
+            feature[~voiced_mask] = 0.0
+
+        training_example[self.field] = feature
+
+
+class MeanVarianceSpeakerNormalization(FeatureProcessor):
+    def __init__(
+        self,
+        field: str,
+        stats_path: Path,
+        speaker_field: str = "speaker",
+        mask_field: Optional[str] = "voiced_mask",
+        fallback_to_default: bool = False,
+    ):
+        """
+        Apply speaker level mean and variance to the input field. Statistics are provided in JSON format, and can be
+        computed using scripts.dataset_processing.tts.compute_feature_stats.py
+
+        Specifically: input[field] = (input[field] + speaker_mean) / speaker_standard_deviation
+
+        Stats file format example for field 'pitch':
+
+        {
+            "default": {
+                "pitch_mean": 100.0,
+                "pitch_std": 50.0,
+            },
+            "speaker1": {
+                "pitch_mean": 110.0,
+                "pitch_std": 45.0,
+            },
+            "speaker2": {
+                "pitch_mean": 105.0,
+                "pitch_std": 30.0,
+            },
+            ...
+        }
+
+        Args:
+            field: Field to apply normalization to.
+            stats_path: JSON file with feature mean and variance.
+            speaker_field: field containing speaker ID string.
+            mask_field: Optional, field in example dictionary with boolean array indicating which values to
+                mask to 0. Defaults to 'voiced_mask', expected to be by pyin pitch estimator.
+            fallback_to_default: Whether to use 'default' feature statistics when speaker is not found in
+                the statistics dictionary.
+        """
+
+        self.field = field
+        self.key_mean = f"{self.field}_mean"
+        self.key_std = f"{self.field}_std"
+        self.speaker_field = speaker_field
+        self.mask_field = mask_field
+        self.fallback_to_default = fallback_to_default
+
+        if not os.path.exists(stats_path):
```

Comment:
This check is rather redundant, `open` will throw more or less the same thing if file does not exist
```
FileNotFoundError: [Errno 2] No such file or directory: <path>
```

Nitpick, not a blocker

Reply:
https://docs.python.org/3/glossary.html#term-EAFP

Reply:
I was not aware until now of EAFP being a standard in Python. But since it is, then it makes sense to not have an explicit assert or try/catch block as the exception is more or less the same, as you pointed out.

Reply:
I'm not talking about catching anything. Please see pseudocode below. I hope redundancy is more visible this way
```
def open(path):
    if not path.exists():
        raise FileNotFoundError
    ...

def do_things(path):
    if not path.exists():
        raise ValueError
    open(path)
```